Event stat boost

LatidoReMe opened the pull request:
__Added boost stat field to event__
===================================================================
Challenge Lives & Band Battles now have a boosted stat! Thus, when adding or editing one of these events, an extra field with the options of Performance, Technique, or Visual will appear!

These fields are technically represented by color in-game, but colors don't look great as icons by themselves so it is a dropdown instead.

The display of the field is pretty simple. it can be found in between boost attribute & boost members in the itemview with the statistics icon being displayed, and then the verbose name of the stat as the value
![image](https://user-images.githubusercontent.com/30684769/45482114-09617680-b702-11e8-9ff4-799a65a24e08.png)

__Added boost stat filter to event__
===================================================================
It'd make sense for users to want to filter by this new field, right? Especially if you wanna prepare for future events on EN, make preds on the next boost stat, etc. etc., a filter for this can come in handy!

Since our sidebar keeps on growing, I decided to make this filter toggleable! If a user is filtering by a Challenge Live or Band Battle, then the field will slide into view! However, if they aren't, then it will remain out of sight.
![image](https://user-images.githubusercontent.com/30684769/44377552-2ebbe580-a4b2-11e8-952e-c39e29bd4399.png) ![image](https://user-images.githubusercontent.com/30684769/45481947-a96ad000-b701-11e8-944c-9a40e97a9603.png)


__Other Notable Things__
===================================================================
If someone happened to set the boost stat for an event and then change the event type afterwards before None-ing the boost stat field, there's no need to worry! I made the save function for Event Form automatically None boost stat if this occurs.

Also I seperated some of the Event/Gacha js since they aren't as similar anymore.

Static Files: https://github.com/MagiCircles/BanGDream-Static/pull/10

db0company commented:
In some languages, `Performance` might not start with the letter "P" so it will not make sense.
Do you think we could just limit it to colored icons? Or other icons used in the game?

violet-r commented:
In my opinion, the stat bubbles will work just fine as the stats are in Katakana/English Loan in the JP game. The field could be translatable in descriptions and it would work fine, I think. But there aren't other assets to borrow that make sense to me.

db0company commented:
What we can do is just show the letters in text, and different letters for different languages (Japanese would be パ for ex), and use CSS to make the colored backgrounds. Would that work for you?

summertriangle-dev commented:
`i_boost_stat` should also be added to the list of API fields.

LatidoReMe commented:
Ok, I tried out what db suggested, and it looks pretty good! It doesn't have the pop-up, but we can add that in later if it's an option.

![](https://cdn.discordapp.com/attachments/320395482247462913/508887701320105985/unknown.png)

Also, I believe I added in api support since triangle mentioned it.

I also added fixes for closing #156 and for making the countdown view consistent with the code and visuals! (since beforehand it wasn't)

If this can get checked out again soon, that'd be great. And thanks for the suggestions!

summertriangle-dev commented on bang/magicollections.py:
```diff
@@ -1171,10 +1171,16 @@ def extra_context(self, context):
 }
 
+EVENT_LIST_ITEM_CUTEFORM['i_boost_stat'] = {
+    'to_cuteform': lambda k, v: list(v)[0],
```
you can grab the first char directly from the string, there's no need to list() it here
